File: app/src/main/java/org/rm3l/ddwrt/resources/PublicIPInfo.java
package org.rm3l.ddwrt.resources;

import android.support.annotation.Nullable;

/**
 * Created by rm3l on 07/09/15.
 */
public class PublicIPInfo {

    public static final String IPIFY_API_RAW = "https://api.ipify.org";
    public static final String IPIFY_API_JSON = "https://api.ipify.org?format=json";

    public static final String ICANHAZIP_HOST = "icanhazip.com";
    public static final int ICANHAZIP_PORT = 80;

    public static final String ICANHAZPTR_HOST = "icanhazptr.com";
    public static final int ICANHAZPTR_PORT = 80;

    @Nullable
    private String ip;

    @Nullable
    public String getIp() {
        return ip;
    }

    public PublicIPInfo setIp(@Nullable String ip) {
        this.ip = ip;
        return this;
    }

    @Override
    public String toString() {
        return "PublicIPInfo {" +
                "ip='" + ip + '\'' +
                '}';
    }

}

File: app/src/main/java/org/rm3l/router_companion/resources/PublicIPInfo.kt
package org.rm3l.router_companion.resources

/**
 * Created by rm3l on 07/09/15.
 */
class PublicIPInfo {

  private var ip: String? = null

  fun getIp(): String? {
    return ip
  }

  fun setIp(ip: String?): PublicIPInfo {
    this.ip = ip
    return this
  }

  override fun toString(): String {
    return "PublicIPInfo {ip='$ip'}"
  }

  companion object {

    val IPIFY_API_RAW = "https://api.ipify.org"
    val IPIFY_API_JSON = "https://api.ipify.org?format=json"

    val ICANHAZIP_HOST = "icanhazip.com"
    val ICANHAZIP_PORT = 80

    val ICANHAZPTR_HOST = "icanhazptr.com"
    val ICANHAZPTR_PORT = 80
  }
}
